pkg/resource/scalable_target/resource.go

```diff
@@ -74,17 +74,37 @@ func (r *resource) Conditions() []*ackv1alpha1.Condition {
 	return r.ko.Status.Conditions
 }
 
+// ReplaceConditions sets the Conditions status field for the resource
+func (r *resource) ReplaceConditions(conditions []*ackv1alpha1.Condition) {
+	r.ko.Status.Conditions = conditions
+}
+
 // SetObjectMeta sets the ObjectMeta field for the resource
 func (r *resource) SetObjectMeta(meta metav1.ObjectMeta) {
 	r.ko.ObjectMeta = meta
 }
 
+// SetStatus will set the Status field for the resource
+func (r *resource) SetStatus(desired acktypes.AWSResource) {
+	r.ko.Status = desired.(*resource).ko.Status
+}
+
 // SetIdentifiers sets the Spec or Status field that is referenced as the unique
 // resource identifier
 func (r *resource) SetIdentifiers(identifier *ackv1alpha1.AWSIdentifiers) error {
-	if identifier.NameOrID == nil {
-		return ackerrors.MissingNameIdentifier
+	if r.ko.Status.ACKResourceMetadata == nil {
+		r.ko.Status.ACKResourceMetadata = &ackv1alpha1.ResourceMetadata{}
 	}
-	r.ko.Spec.ResourceID = identifier.NameOrID
+	r.ko.Status.ACKResourceMetadata.ARN = identifier.ARN
```

Comment:
Spec for adopted resource has optional `nameOrID` OR `arn` fields - https://github.com/aws-controllers-k8s/runtime/blob/86654532edbd5e5cf99d7f89c2aa561bce810f14/apis/core/v1alpha1/identifiers.go#L17

The user can specify one or the other. Depending on what the describe call expects, it will use that value to populate the CR spec.

Reply:
based on further discussion with Nick,

it probably works like this:
1. The adoption reconciler creates an empty ScalableTarget CR
2. We set ARN = identifier.ARN (which is an empty string) - so we still have an empty ScalableTarget CR
3. We call ReadOne, which under the hood calls ReadMany with an empty ResourceIds list
4. The result of the ReadOne is the first element in the ReadMany call and there is an element in result **because resourceId is an optional parameter**
5. We set the CR to that result. so it might be some random scalable target in the service namespace, not necessarily the one it was supposed to adopt.

Recommend the following:
1. add an assert for that spec.resourceId in the test
2. work on [fixing this](https://github.com/aws-controllers-k8s/applicationautoscaling-controller/blob/cff66844fa3b80077b3e3cdaeb3c9bc1d39a7eb2/pkg/resource/scalable_target/sdk.go#L143) as must have before release

Reply:
for 2 -> https://github.com/aws-controllers-k8s/code-generator/pull/155

this should have helped

Reply:
PR linked above will not work out of the box because it will be looking for `ScalableTargetIds` or `ScalableTargetId` [ref](https://github.com/aws-controllers-k8s/code-generator/pull/155/files#diff-5abe25faaf60c3e75e2732770b127492399b415883df4780b96bb6814118388aR35-R36)

but for this resource required field is actually `resourceIds`

Reply:
Yea, also need to change the field identified as arn/name below. Both ScalableTarget and ScalingPolicy have bugs atm.